src/main/java/edu/ucsb/cs156/gauchoride/controllers/RideRequestController.java

```diff
@@ -0,0 +1,132 @@
+        String body = mapper.writeValueAsString(rideRequests);
+        return ResponseEntity.ok().body(body);
+    }
+
+    @ApiOperation(value = "Create a new ride request")
+    @PreAuthorize("hasRole('ROLE_RIDER')")
+    @PostMapping("/post")
+    public RideRequest postRideRequest(
+        @ApiParam(name="userid", type="Long", value="userid of the rider", example="123", required=true)  @RequestParam Long userid,
+        @ApiParam(name="day", type="String", value="day, one of [Monday, Tuesday,..., Sunday]", example="Monday", required=true) @RequestParam String day,
+        @ApiParam(name="fullName", type="String", value="name of the rider", example="Joe", required=true) @RequestParam String fullName,
+        @ApiParam(name="course", type="String", value="course to attend", example="CS 156", required=true) @RequestParam String course,
+        @ApiParam(name="startTime", type="String", value="start time, format: HH:MM XM", example="12:00 AM", required=true) @RequestParam String startTime,
+        @ApiParam(name="stopTime", type="String", value="stop time, format: HH:MM XM", example="12:00 AM", required=true) @RequestParam String stopTime,
+        @ApiParam(name="building", type="String", value="destination building", example="Phelps", required=true) @RequestParam String building,
+        @ApiParam(name="room", type="String", value="destination room", example="EUCR, 3525", required=true) @RequestParam String room,
+        @ApiParam(name="pickup", type="String", value="pickup location", example="San Rafael Hall", required=true) @RequestParam String pickup
+        )
+        throws JsonProcessingException {
+
+        RideRequest rideRequest = new RideRequest();
+        rideRequest.setUserid(userid);
+        rideRequest.setDay(day);
+        rideRequest.setFullName(fullName);
+        rideRequest.setCourse(course);
+        rideRequest.setStartTime(startTime);
+        rideRequest.setStopTime(stopTime);
+        rideRequest.setBuilding(building);
+        rideRequest.setRoom(room);
+        rideRequest.setPickup(pickup);
+
+        RideRequest saved = rideRequestRepository.save(rideRequest);
+
+        return saved;
+    }
+
+    @ApiOperation(value = "Delete a ride request")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_RIDER')")
+    @DeleteMapping("/delete")
+    public Object deleteRideRequest_Admin(
+            @ApiParam("id") @RequestParam Long id) {
+              RideRequest rideRequest = rideRequestRepository.findById(id)
+          .orElseThrow(() -> new EntityNotFoundException(RideRequest.class, id));
+
+          rideRequestRepository.delete(rideRequest);
+
+        return genericMessage("Ride request with id %s deleted".formatted(id));
+    }
+
+    @ApiOperation(value = "Update a ride request")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_RIDER')")
+    @PutMapping("/put")
+    public RideRequest updateRideRequest(
+            @ApiParam("id") @RequestParam Long id,
```

Comment:
Same deal as above with DELETE.  Riders should only be able to update their own requests, while admins should be able to update any request.

The code to do this isn't complicated.